src/Arcus.Security.Core/Caching/CachedSecretProvider.cs

```diff
@@ -89,6 +89,7 @@ public CachedSecretProvider(ISecretProvider secretProvider) :
         /// <exception cref="ArgumentException">The name must not be empty</exception>
         /// <exception cref="ArgumentNullException">The name must not be null</exception>
         /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        [Obsolete("Will be removed in v3 in favor of solely using " + nameof(GetSecret) + " instead")]
```

Comment:
When making these methods obsolete, shouldn't we (in the same PR?) also already introduce the implicit cast operator that allows to convert from `Secret` to `string` ?